src/main/java/ch/njol/skript/structures/StructEvent.java

```diff
@@ -46,8 +47,25 @@ public class StructEvent extends Structure {
 	@SuppressWarnings("ConstantConditions")
 	public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResult, EntryContainer entryContainer) {
 		String expr = parseResult.regexes.get(0).group();
-		if (!parseResult.tags.isEmpty())
-			getParser().getData(EventData.class).priority = EventPriority.valueOf(parseResult.tags.get(0).toUpperCase(Locale.ENGLISH));
+
+		EventData data = getParser().getData(EventData.class);
+		// clear old data values
+		data.behavior = null;
+		data.priority = null;
+
+		if (parseResult.hasTag("uncancelled")) {
+			data.behavior = ListeningBehavior.UNCANCELLED;
+		} else if (parseResult.hasTag("cancelled")) {
+			data.behavior = ListeningBehavior.CANCELLED;
+		} else if (parseResult.hasTag("any")) {
+			data.behavior = ListeningBehavior.ANY;
+		}
+
+		if (parseResult.hasTag("priority")) {
```

Comment:
I'm not sure this is necessary. There would only ever be at most two tags (listening behavior and priority), right?

Reply:
Well the only other option I can think of is checking for all 6 priority tags individually.
Because if there's only one tag supplied, it's impossible to know whether it's a priority or a listening behavior without checking them all.

Reply:
I mean it would still be the last tag right?